src/AWSMetadataUtilities.jl

```diff
@@ -321,64 +354,107 @@ function _generate_high_level_definition(
     optional_parameters::AbstractDict,
     documentation::String
 )
-    required_param_keys = collect(keys(required_parameters))
-
     operation_definition = """
-    \"\"\"
+    $(repeat('"', 3))
         $name()
 
-    $documentation
+    $documentation\n
     """
 
     # Add in the required parameters if applicable
     if !isempty(required_parameters)
-        operation_definition *= """
+        operation_definition *= "Required Parameters\n"
 
-        Required Parameters
-        $(json(required_parameters, 2))"""
+        for required_parameter in required_parameters
+            operation_definition *= "$(required_parameter[1]) => $(required_parameter[2]["documentation"])\n"
+        end
+
+        operation_definition *= "\n"
     end
 
     # Add in the optional parameters if applicable
     if !isempty(optional_parameters)
-        operation_definition *= """
+        operation_definition *= "Optional Parameters\n"
 
-        Optional Parameters
-        $(json(optional_parameters, 2))"""
+        for optional_parameter in optional_parameters
+            operation_definition *= "$(optional_parameter[1]) => $(optional_parameter[2])\n"
+        end
     end
 
     operation_definition *= repeat('"', 3)
 
+    # Get a list of Julia friendly parameter keys
+    # Replace all hyphens with underscore to be Julia friendly
+    required_param_keys_clean = replace.(collect(keys(required_parameters)), "-"=>"_")
+
+    # Filter out any required parameters which are in the URI
+    required_parameters = filter(p -> (p[2]["location"] != "uri"), required_parameters)
+
+    # Filter all header parameters
+    header_parameters = filter(p -> (p[2]["location"] == "header"), required_parameters)
+    required_parameters = setdiff(required_parameters, header_parameters)
+
+    # "par-am"=>par_am
+    definition_required_parameters = ["\"$(p[1])\"=>$(replace(p[1], "-"=>"_"))" for p in required_parameters]
+    header_parameters = ["\"$(p[1])\"=>$(replace(p[1], "-"=>"_"))" for p in header_parameters]
```

Comment:
```suggestion
    # "par-am" => par_am
    definition_required_parameters = ["\"$(p[1])\"=>$(replace(p[1], "-" => "_"))" for p in required_parameters]
    header_parameters = ["\"$(p[1])\"=>$(replace(p[1], "-" => "_"))" for p in header_parameters]
```
Consider making a function out of `replace(x, "-" => "_")`